src/api.v2/helpers/s3/patchCellSetsObject.js

```diff
@@ -0,0 +1,39 @@
+const jsonMerger = require('json-merger');
+
+const bucketNames = require('./bucketNames');
+const getObject = require('./getObject');
+const putObject = require('./putObject');
+
+const validateRequest = require('../../../utils/schema-validator');
+
+const patchCellSetsObject = async (experimentId, patch) => {
+  const initialCellSet = await getObject({
+    Bucket: bucketNames.CELL_SETS,
+    Key: experimentId,
+  });
+
+  const { cellSets: cellSetsList } = initialCellSet;
+
+  /**
+   * The $remove operation will replace the element in the array with an
+   * undefined value. We will therefore remove this from the array.
+   *
+   * We use the $remove operation in the worker to update cell clusters,
+   * and we may end up using it in other places in the future.
+   */
```

Comment:
This is actually not true, the `$remove` removes when used well, the problem was that the first remove (the one used in worker and pipeline) is not written well, so we need to remove the `undefined`s it leaves behind.

As an example, the `.filter((x) => x !== undefined);` works only at root level, so the deletes we send from the UI for one cellSet within a cellClass would still leave `undefined` if this were true.

Reply:
And also the `$remove` is used in the worker, pipeline and UI as well, so I would suggest updating this comment to explain the fact that it's a bug in the json merger command in worker and pipeline. (if it isn't fixed in this ticket)

Reply:
After discussion, this will not be done in this PR, but with in another ticket which adds v2 functionality to the worker and pipeline